compiler/src/iree/compiler/InputConversion/StableHLO/StableHLOToLinalgRandom.cpp

```diff
@@ -0,0 +1,591 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// Implements logic for lowering StableHLO random number generation to Linalg
+// dialect.
+
+#include "iree/compiler/InputConversion/StableHLO/LegalizeToLinalgUtils.h"
+#include "iree/compiler/InputConversion/StableHLO/Rewriters.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/Support/LogicalResult.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "stablehlo/dialect/StablehloOps.h"
+
+namespace mlir::iree_compiler::stablehlo {
+namespace {
+namespace stablehlo = mlir::stablehlo;
+
+class ArithOp {
+ public:
+  ArithOp(OpBuilder b, Location l, Value v) : builder(b), loc(l), value(v) {}
```

Comment:
This also feels out of place. Ops are typically defined in tablegen, but this class is a bunch of helper methods... can those be upstreamed in the arith dialect? Or should this be renamed to `ArithOpUtilBuilder` (or something that gives me context for why it exists)?

Reply:
Yes, this is more of a helper to build existing ops. I will rename it.